### ISSUE TITLE
Release reproducible training code and model weights

### ISSUE THREAD

**Issue by cs-xyzhang** [liuhaogeng/Anchor-Former#1]
Dear authors,

Thank you for your work. I have followed the hyperparameter settings described in Table 6 of the paper and the details provided in the text, using 8 A100 GPUs for reproduction. However, after multiple attempts, I still cannot reproduce results close to those reported in Tables 1 and 2. The results I obtained are even worse than LLaVA-1.5.

Could you please provide the pretrained model weights and training code necessary to reproduce the results presented in the paper? I believe that, as an MLLM work published in the top conference NIPS, this is a reasonable request. 

Thank you.

**Comment by liuhaogeng** [comment]
Thank you for your attention! We’ve recently been busy with the ICLR rebuttal. However, we retrained a 7B model with a token count of 145 using our previous code yesterday, and the test results are as follows. They show almost no difference compared to our paper.

(Please note that in multimodal large language models (MLLMs), fluctuations within a range of 1 are quite normal. For example, when I ran the LLAVA code previously, the results were 0.7 lower than the reported numbers in the paper. Additionally, two questions in the MME benchmark have known issues, which were automatically counted as incorrect in prior evaluations. As a result, a difference of around 40 in MME precision is also expected.)

Could you kindly share the results of your current model with us?
<img width="676" alt="1732068922496" src="https://github.com/user-attachments/assets/8db5f025-90a9-4899-8764-251e591f29a8">
<img width="726" alt="1732068911459" src="https://github.com/user-attachments/assets/66fcf4dc-aed6-47f8-bf28-9d99ee2f3547">
<img width="686" alt="1732068892518" src="https://github.com/user-attachments/assets/80c4b014-18af-466a-92cb-d2462beb295f">

**Comment by liuhaogeng** [comment]
By the way, for quick experiment, we set the max_token_length=1024，when IFT.

**Comment by liuhaogeng** [comment]
Thanks for your question! We have found the bug! As you know, the data flow are vit->Anchor Former->mlp->llm. Here if you set gradient ckpt=True， the trainer of hugging face will automaticaly set all of the sub module's gradient_checkpointing attribute to True, this will result in Anchor Former not updated during training. We recommend you to remove the grad for the structure (LLaVA/llava/model/multimodal_projector/builder.py, ), this will fix the performance. Just remove this
<img width="440" alt="0bf9f05fb604463b5ee538b0e8f9d95" src="https://github.com/user-attachments/assets/f0a7b650-d9dc-4d9e-8cac-89bc453cd9e4">
